frontend/js/pages/BattleDetail.js

```diff
@@ -25,16 +26,16 @@ const Wrapper = styled.section`
   background: linear-gradient(to right, rgb(197, 230, 236), rgb(239, 187, 230));
 `;
 
-function BattleDetail(props) {
-  const { loading } = props.battle;
-  const { error } = props.battle;
+const BattleDetail = (props) => {
   const { id } = useParams();
+  const { battles, loading, error } = props;
+  const { user } = props.user;
   useEffect(() => {
-    props.getCurrentUser();
-    props.getBattle(id);
+    if (battles.length === 0 || battles.battle[id]) {
```

Comment:
I might be misunderstanding something but I'm not sure what `battles.battle[id]` is doing here, perhaps it should be `!battles.battle[id]`, as in "is battle is not yet in the store, fetch it`?

Reply:
Yes, just fixed it :)